lib/cocoapods/installer.rb

```diff
@@ -350,7 +350,7 @@ def lock_pod_sources
     def determine_dependency_product_types
       aggregate_targets.each do |aggregate_target|
         aggregate_target.pod_targets.each do |pod_target|
-          pod_target.host_requires_frameworks = aggregate_target.requires_frameworks?
+          pod_target.host_requires_frameworks ||= aggregate_target.requires_frameworks?
```

Comment:
As the deduplication doesn't care about the build type, this will opt-in all `PodTarget` to frameworks, which are shared as a deduplicated dependency with a target definition, which is opted-in to use frameworks. Otherwise it would seem to be less deterministic, whether a `PodTarget` is build as framework or not.